Source/WTF/wtf/HashMap.h

```diff
@@ -37,7 +37,7 @@ template<typename T> struct KeyValuePairKeyExtractor {
     static const typename T::KeyType& extract(const T& p) { return p.key; }
 };
 
-template<typename KeyArg, typename MappedArg, typename HashArg, typename KeyTraitsArg, typename MappedTraitsArg, typename TableTraitsArg>
+template<typename KeyArg, typename MappedArg, typename HashArg, typename KeyTraitsArg, typename MappedTraitsArg, typename TableTraitsArg, ShouldValidateKey validateKey>
```

Comment:
I prefer a naming scheme where the variable matches the type: `ShouldValidateKey shouldValidateKey`.

It's more textually regular, if you're searching the project. It's usually the best name, since you gave the type a good name. And in this case, `validateKey` sounds like it might be a function you call to validate the key.

Minor point, but probably worth fixing.